torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -283,6 +289,198 @@ inline void errorIfCapturingNonCapturableNCCL(c10::cuda::CaptureStatus status) {
 
 } // namespace
 
+namespace {
+std::string write_pickle(const c10::IValue& v) {
+  std::vector<char> result;
+  {
+    auto writer = [&](const char* data, size_t size) {
+      result.insert(result.end(), data, data + size);
+    };
+    torch::jit::Pickler pickler(
+        writer, nullptr, nullptr, nullptr, nullptr, false);
+    pickler.protocol();
+    pickler.pushIValue(v);
+    pickler.stop();
+  }
+  return std::string(result.begin(), result.end());
+}
+c10::Dict<c10::IValue, c10::IValue> new_dict() {
+  return c10::Dict<c10::IValue, c10::IValue>(
+      c10::AnyType::get(), c10::AnyType::get());
```

Comment:
Nit
C++ should auto-create an object, shouldn't it, i.e. 
```cpp
std::unordered_map<int, int> foo() {
    return {{1, 2}};
}
```
```suggestion
  return {{c10::AnyType::get(), c10::AnyType::get()}};
```

Reply:
ctor's are marked explicit, so it doesn't work.